draft-ietf-jsonpath-base.md

```diff
@@ -1115,11 +1115,10 @@ logical-and-expr    = basic-expr *(S "&&" S basic-expr)
 basic-expr          = paren-expr /
                       comparison-expr /
                       test-expr
-test-expr           = [logical-not-op S] filter-path /
-                         ; path existence or non-existence
-                      function-expr ; OptionalBooleanType or subtype
-filter-path         = rel-path / json-path /
-                      function-expr ; OptionalNodesType or subtype
+test-expr           = [logical-not-op S]
+                      (filter-path / ; path existence or non-existence
+                       function-expr) ; OptionalBooleanType or subtype
```

Comment:
The above function-expr is now the only one in test-expr, so it's not limited to OptionalBooleanType or subtype. It needs to include OptionalNodesType or subtype too.

Approving as I expect these comments to reworked soon.

Reply:
I just added the notation ST(T), which makes it easier to compactly speak about these things.
Also, some editorial cleanup.